File: src/main/java/plus/maa/backend/common/annotation/CurrentUser.java
package plus.maa.backend.common.annotation;

import org.springframework.security.core.annotation.AuthenticationPrincipal;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
/**
 * @author john180
 */
@Target({ElementType.PARAMETER})
@Retention(RetentionPolicy.RUNTIME)
@AuthenticationPrincipal
public @interface CurrentUser {
}

File: src/main/kotlin/plus/maa/backend/common/annotation/CurrentUser.kt
package plus.maa.backend.common.annotation

import org.springframework.security.core.annotation.AuthenticationPrincipal

/**
 * @author john180
 */
@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
@AuthenticationPrincipal
annotation class CurrentUser
